charts/hivemq-platform/tests/hivemq_services_test.yaml

```diff
@@ -184,42 +184,1196 @@ tests:
           path: spec.services
           content:
             metadata:
-              name:  hivemq-test-hivemq-platform-mqtt-1884
+              name: hivemq-test-hivemq-platform-mqtt-1884
+            spec:
+              ports:
+                - name: mqtt-1884
+                  targetPort: mqtt-1884
+                  port: 1884
+
+  - it: with a Control Center service enabled
+    set:
+      services:
+        - type: control-center
+          exposed: true
+          containerPort: 8081
+    asserts:
+      - exists:
+          path: spec.statefulSet.spec.template.spec.containers[0].ports[0]
+      - contains:
+          path: spec.statefulSet.spec.template.spec.containers[0].ports
+          content:
+            name: cc-8081
+            containerPort: 8081
+      - exists:
+          path: spec.services
+      - contains:
+          path: spec.services
+          content:
+            metadata:
+              name: hivemq-test-hivemq-platform-cc-8081
+            spec:
+              ports:
+                - name: cc-8081
+                  targetPort: cc-8081
+                  port: 8081
+              sessionAffinity: ClientIP
+
+  - it: with duplicated containerPorts in enabled services, then services are created successfully
+    set:
+      services:
+        - type: mqtt
```

Comment:
I'm not 100% sure if this scenario makes sense, since the same port should never be able to serve two different service types.

I would see this scenario for the same type though, but then you need to use a custom service name. For example, `mqtt-internal` and `mqtt-external` using the same port (e.g. internal to monitor the same listener, but without load-balancer).

Reply:
I think you're right. The main problem I see is because of the listeners defined for the broker config. So with a values like:

```yaml
services:
  - type: mqtt
    exposed: true
    containerPort: 1883
  - type: control-center
    exposed: true
    containerPort: 1883
```

The broker configuration generated will look like:
```xml
...
  <listeners>
    <tcp-listener>
      <port>1883</port>
      <bind-address>0.0.0.0</bind-address>
    </tcp-listener>
  </listeners>
...
  <control-center>
    <listeners>
      <http>
        <port>1883</port>
        <bind-address>0.0.0.0</bind-address>
      </http>
    </listeners>
  </control-center>
  ...
```

Hence, the broker won't start up:

```
ERROR - Could not start TCP Listener on port 1883 and address 0.0.0.0. Is it already in use?
ERROR - Could not bind any listener. Stopping HiveMQ.
```

In a second test with the following values, the same broker configuration will be generated, hence the broker still not coming up:

```yaml
services:
  - type: mqtt
    name: my-mqtt-1
    exposed: true
    containerPort: 1883
  - type: control-center
    name: my-mqtt-2
    exposed: true
    containerPort: 1883
```

In a 3rd test though, I tried with the following values:
```yaml
services:
  - type: mqtt
    name: my-mqtt-1
    exposed: true
    containerPort: 1883
  - type: mqtt
    name: my-mqtt-2
    exposed: true
    containerPort: 1883
```

And the generated broker configuration file has duplicated listeners:
```xml
  <listeners>
    <tcp-listener>
      <port>1883</port>
      <bind-address>0.0.0.0</bind-address>
    </tcp-listener>
    <tcp-listener>
      <port>1883</port>
      <bind-address>0.0.0.0</bind-address>
    </tcp-listener>
  </listeners>
```

So we should probably:
1. Validate there is no combination in a service with duplicated container port and different service type.
2. For same service type, same container port but different service name, it should be valid and the broker configuration should not create a duplicated listener.